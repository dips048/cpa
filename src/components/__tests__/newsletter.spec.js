import React from "react"
import renderer from "react-test-renderer"
import Newsletter from "../newsletter"
import { render } from "@testing-library/react"

describe("Newsletter", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Newsletter/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("should render given `title`" , () => {
    const { getByText, getByTestId } = render(<Newsletter />)
    expect(getByText("Newsletter")).toBeInTheDocument()
    expect(getByText("Subscribe to our newsletter to receive news, updates, and valuable tips.")).toBeInTheDocument()
    expect(getByTestId("fa-icon")).toBeInTheDocument()
  })
})
