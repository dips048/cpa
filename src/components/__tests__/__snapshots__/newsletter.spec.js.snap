// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsletter renders correctly 1`] = `
<div
  className="container"
>
  <h2>
    Newsletter
  </h2>
  <p>
    Subscribe to our newsletter to receive news, updates, and valuable tips.
  </p>
  <div
    className="mx-auto"
  >
    <form
      className="col-sm-8 mx-auto d-inline"
    >
      <input
        name="email"
        placeholder="Email"
        type="text"
      />
      <button
        aria-label="submit"
        className="btn btn-primary"
        name="submit"
        type="submit"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-rss "
          data-icon="rss"
          data-prefix="fas"
          data-testid="fa-icon"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M25.57 176.1C12.41 175.4 .9117 185.2 .0523 198.4s9.173 24.65 22.39 25.5c120.1 7.875 225.7 112.7 233.6 233.6C256.9 470.3 267.4 480 279.1 480c.5313 0 1.062-.0313 1.594-.0625c13.22-.8438 23.25-12.28 22.39-25.5C294.6 310.3 169.7 185.4 25.57 176.1zM32 32C14.33 32 0 46.31 0 64s14.33 32 32 32c194.1 0 352 157.9 352 352c0 17.69 14.33 32 32 32s32-14.31 32-32C448 218.6 261.4 32 32 32zM63.1 351.9C28.63 351.9 0 380.6 0 416s28.63 64 63.1 64s64.08-28.62 64.08-64S99.37 351.9 63.1 351.9z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </form>
  </div>
</div>
`;
